app/import_functions.go

```diff
@@ -960,123 +960,171 @@ func (a *App) importAttachment(data *AttachmentImportData, post *model.Post, tea
 	return fileInfo, nil
 }
 
-func (a *App) importPost(data *PostImportData, dryRun bool) *model.AppError {
-	if err := validatePostImportData(data, a.MaxPostSize()); err != nil {
-		return err
+type PostAndData struct {
+	Post           *model.Post
+	PostData       *PostImportData
+	DirectPostData *DirectPostImportData
+	Team           *model.Team
+}
+
+func (a *App) importMultiplePosts(data []*PostImportData, dryRun bool) *model.AppError {
+	if len(data) == 0 {
+		return nil
+	}
+
+	for _, postData := range data {
+		if err := validatePostImportData(postData, a.MaxPostSize()); err != nil {
+			return err
+		}
 	}
 
 	// If this is a Dry Run, do not continue any further.
 	if dryRun {
 		return nil
 	}
 
-	team, err := a.Srv.Store.Team().GetByName(*data.Team)
+	teams := make(map[string]*model.Team)
+	channels := make(map[string]*model.Channel)
```

Comment:
It's not very clear to me the purpose of these channels and users maps. They seem to be just populated and not used.

Is it just to ensure that the users and channels exist in the DB ? Maybe put a comment to clarify that ?

Reply:
There was some errors using it.

Reply:
Actually, rethinking it, it wasn't wrong, the idea behind that teams/channels is to cache the channels while we are importing posts avoiding to get the same channel, user or team multiple times.

Reply:
This defines a set of maps containing users and channels by name (empty at the beginning), then it loop over the set of posts that you want to import, for each loop iteration it tries to get from that map the channel and user associated to the post, if is not present in the map, it get the User/Channel from the database, and store it in the map. The next iteration of the loop is going to search in the map, if the channel/user is already loaded it uses it, if not, it request it and save it in the map. At the end, is a cache inside the method.